File: Backend/app/schemas/news.py
from pydantic import BaseModel
from typing import List
from datetime import datetime


class NewsResponse(BaseModel):
    id: int
    title: str
    summary: str
    source: str
    source_url: str
    image_url: str
    category: str
    tags: List[str] = []
    read_time: str
    published_at: datetime

    class Config:
        from_attributes = True


class NewsListResponse(BaseModel):
    articles: List[NewsResponse]
    total: int
    page: int
    limit: int

File: Frontend/src/types/news.ts
export type NewsArticle = {
  id: string;
  title: string;
  summary: string;
  content: string;
  source: string;
  author: string;
  url: string;
  imageUrl?: string;
  category: NewsCategory;
  tags: string[];
  publishedAt: string;
  readTime: number;
};

export type NewsCategory = 'AI' | 'Web' | 'Mobile' | 'DevOps' | 'Programming' | 'Tech' | 'Tools' | 'Data Science' | 'cloud' | 'Security' | 'other';

export type DailyUpdate = {
  date: string;
  articles: NewsArticle[];
  aiTools: AITool[];
  trendingTopics: string[];
};

export type AITool = {
  name: string;
  description: string;
  url: string;
  category: string;
  pricing: 'Free' | 'Freemium' | 'Paid';
  rating: number;
};